blog/_posts/2022-01-17-scala-3.1.1-released.md

```diff
@@ -0,0 +1,71 @@
+- Scaladoc can correctly handle symbolic class names, such as `/` or `\` ([PR #13827](https://github.com/lampepfl/dotty/pull/13827)).
+- Scaladoc documentation has a new, improved UI.
+
+For a comprehensive list of fixes, take a look at the full changelog on GitHub. // TODO: link
+
+## 3.1.2-RC1 and forward compatibility improvements
+
+Scala 3 has excellent backward compatibility guarantees between the minor versions. On the other hand, after the recent release of Scala 3.1, we can see that the libraries should be really cautious with updating the compiler version, as it is forcing the bump on every user of that library. We do not want library authors to be stuck on old versions of the compiler as that would mean that they are locked out of many bugfixes, or we would need to spend enormous effort on backporting every bugfix to all past versioning lines.
+
+We are sure we want to mitigate those difficulties. We already have implemented the first and most important part of the solution and have it ready for public testing. Beginning in Scala 3.1.2-RC1, that was just released, compiler is able to generate output that can be consumed by older versions. To enable that, you need to specify a minimum version of the language using the experimental `-Yscala-release` flag.
+
+So, for example, compiling your library using Scala 3.1.2-RC2 with `-Yscala-release:3.0` flag, you make it possible for people still using Scala 3.0 to depend on your library. This has one limitation, you won't be able to use any symbol added to the standard library after 3.0. They are marked in the code and in the documentation by the internal `@since` annotation that was also added in this version.
+
+Those restrictions are not final, and in the future, we may decide to allow using some functions from the newer versions of the standard library and to make the `@since` annotation public.
+
+One important note is that the `-Yscala-release` flag is experimental and not yet available in any stable release of the compiler. It means the build tools, such as SBT or Mill, are not yet aware of its existence. Of course, you can declare usage of the flag in your build file, and the compiler will respect it. However, there may still be some hiccups, like generated artifacts depending on the incorrect version of the stdlib. We will do our best to help fix those minor issues before the stable release of 3.1.1.
+
+If you are interested in other changes brought by 3.1.2-RC1, there is, as always, a changelog on GitHub.  // TODO: link
+
+## Contributors
+
+// TODO: contributors list
+
+## Library authors: Join our community build
+
+Scala 3 has a set of widely-used community libraries that are built against every nightly Scala 3 snapshot.
+Join our [community build](https://github.com/lampepfl/dotty/tree/master/community-build)
+to make sure that our regression suite includes your library.
```

Comment:
I doubt this is sufficiently important to repeat in every release announcement.

Reply:
Sure, removed.